src/runtime-rs/Makefile

```diff
@@ -88,12 +88,19 @@ HYPERVISOR_FC = firecracker
 HYPERVISOR_QEMU = qemu
 HYPERVISOR_CLH = cloud-hypervisor
 
+##VAR DRAGONBALL=yes|no Define whether dragonball is compiled into runtime-rs.
+DRAGONBALL := yes
 
 DEFAULT_HYPERVISOR ?= $(HYPERVISOR_DB)
 
 ##VAR HYPERVISOR=<hypervisor_name> List of hypervisors this build system can generate configuration for.
 HYPERVISORS := $(HYPERVISOR_DB) $(HYPERVISOR_ACRN) $(HYPERVISOR_FC) $(HYPERVISOR_QEMU) $(HYPERVISOR_CLH)
 
+# If DRAGONBALL is not set to 'yes', set the default hypervisor to QEMU.
```

Comment:
Why set default hypervisor to QEMU ?

Reply:
Since the DEFAULT_HYPERVISOR controls which hypervisor configuration file is installed during make install, if the default hypervisor isn't modified, configuration.toml will link to the dragonball configuration by default. This will cause the system to fail at startup when dragonball hasn't been compiled. Given that qemu currently has the highest level of completeness in runtime-rs (as far as I know), I've set qemu as the default hypervisor

Reply:
IMHO, `DRAGONBALL` is only available when hypervisor is set to Dragonball. Otherwise, ignore it. Wdyt?

Reply:
Currently, `DRAGONBALL` is set to 'yes' by default. If `DRAGONBALL` is set to 'no',  the modification of the `DEFAULT_HYPERVISOR` will only take effect when `make install DRAGONBALL=no` is executed. In fact, even without modifying the `DEFAULT_HYPERVISOR`, users can also use a non-DRAGONBALL hypervisor to start Kata by manually linking or copying the configuration files of other hypervisors to the configuration path. Is it necessary to remove the part that modifies the` DEFAULT_HYPERVISOR`?

Reply:
What happens when executing `make install DRAGONBALL=no DEFAULT_HYPERVISOR=dragonball`? `DRAGONBALL` is a little bit ambiguous from my perspective.

Reply:
make install will The make install process does the following: it installs the shim to the /usr/local/bin directory, installs the configs for different hypervisors to the CONFIG_PATH, and finally links the DEFAULT_HYPERVISOR_CONFIG to (DESTDIR)/$(CONFIG_PATH) based on the DEFAULT_HYPERVISOR. Therefore, the role of DEFAULT_HYPERVISOR is essentially to determine the DEFAULT_HYPERVISOR_CONFIG. If we specify both DRAGONBALL=no and DEFAULT_HYPERVISOR=dragonball, the DEFAULT_HYPERVISOR will override the previous modification, and in the end, the configuration file for DRAGONBALL will be linked to (DESTDIR)/$(CONFIG_PATH).

Reply:
So `DRAGONBALL ` is used to indicate whether the Dragonball is built-in, and `DEFAULT_HYPERVISOR` is used to link config. Am I right?

Reply:
> So `DRAGONBALL` indicates whether the Dragonball is built-in, and `DEFAULT_HYPERVISOR` is used to link config. Am I right?


Yes, that's indeed the case.

Reply:
Thanks for confirmation. IMHO, the `DRAGONBALL` can be renamed to `BUILT_IN_DRAGONBALL` or some similar to eliminate the ambiguous semantics.

Then, there are there cases to take care of (don't go through the cases where the variables are not set here):

1. `DEFAULT_HYPERVISOR=dragonball`, `BUILT_IN_DRAGONBALL=yes`: build shim with Dragonball in built-in mode.
2. `DEFAULT_HYPERVISOR=dragonball`, `BUILT_IN_DRAGONBALL=no`: build shim and Dragonball respectively.
3. `DEFAULT_HYPERVISOR` is not dragonball: Ignore the value of `BUILT_IN_DRAGONBALL`.